crates/prover/src/runner.rs

```diff
@@ -411,9 +408,27 @@ where
                 .unwrap();
 
             let mut sequencer_commitments = Vec::<SequencerCommitment>::new();
-            let mut zk_proofs = Vec::<Proof>::new();
 
-            self.extract_relevant_l1_data(l1_block, &mut sequencer_commitments, &mut zk_proofs);
+            let mut da_data = self.da_service.extract_relevant_blobs(l1_block);
+            // if we don't do this, the zk circuit can't read the sequencer commitments
+            da_data.iter_mut().for_each(|blob| {
+                blob.full_data();
+            });
+            da_data.iter_mut().for_each(|tx| {
+                let data = DaData::try_from_slice(tx.full_data());
+                // Check for commitment
+                if tx.sender().as_ref() == self.sequencer_da_pub_key.as_slice() {
+                    if let Ok(DaData::SequencerCommitment(seq_com)) = data {
+                        sequencer_commitments.push(seq_com);
+                    } else {
+                        tracing::warn!(
+                            "Found broken DA data in block 0x{}: {:?}",
+                            hex::encode(l1_block.hash()),
+                            data
+                        );
+                    }
```

Comment:
imo it would look better if we had a function `extract_sequencer_commitments`

Reply:
Done